ui-components/src/helpers/filters.ts

```diff
@@ -1,70 +1,209 @@
 import {
   EnumListingFilterParamsComparison,
   ListingFilterKeys,
-  ListingFilterParams,
 } from "@bloom-housing/backend-core/types"
 import { ParsedUrlQuery } from "querystring"
+import { SelectOption } from "./formOptions"
+import { t } from "./translator"
+import { AvailabilityFilterEnum } from "@bloom-housing/backend-core/dist"
+import Listing from "@bloom-housing/backend-core/dist/src/listings/entities/listing.entity"
 
-function getComparisonForFilter(filterKey: ListingFilterKeys) {
-  switch (filterKey) {
-    case ListingFilterKeys.name:
-    case ListingFilterKeys.status:
-    case ListingFilterKeys.leasingAgents:
-      return EnumListingFilterParamsComparison["="]
-    case ListingFilterKeys.bedrooms:
-    case ListingFilterKeys.minRent:
-      return EnumListingFilterParamsComparison[">="]
-    case ListingFilterKeys.maxRent:
-      return EnumListingFilterParamsComparison["<="]
-    case ListingFilterKeys.zipcode:
-      return EnumListingFilterParamsComparison["IN"]
-    case ListingFilterKeys.seniorHousing:
-    case ListingFilterKeys.availability:
-    case ListingFilterKeys.ami:
-      return EnumListingFilterParamsComparison["NA"]
-    default: {
-      const _exhaustiveCheck: never = filterKey
-      return _exhaustiveCheck
+export const COMMUNITY_TYPE = "communityType"
```

Comment:
I added a `FrontendFilterKey` enum, however, I am not able to set it by expanding and assigning the ListingFilterKeys because those get stored in a variable which I wasn't able to use as a type later on.

I'm not sure I understand why we want to use Partial though. If a filter exists in FrontendFilterKeys, I think it should be initialized and assigned a comparison operator. I think allowing filters to be missing can lead to runtime errors down the line.

Reply:
Mmm yeah you're right re `Partial`. We aways want the filter list to have `FrontendListingFilter`s for every filter, even if they're blank, which is what we get in your init method.

I was thinking we'd need `Partial` so we could set filters one at a time in places like the eligibility questionnaire, instead of all at once, but that's not actually how this filter code works.

Reply:
Okay, going to resolve this!